cadquery/occ_impl/shapes.py

```diff
@@ -1150,6 +1199,67 @@ class Solid(Shape, Mixin3D):
     a single solid
     """
     
+    @classmethod
+    def interpPlate(cls, surf_edges, surf_pts, thickness, Degree=3, NbPtsOnCur=15, NbIter=2, Anisotropie=False, Tol2d=0.00001, Tol3d=0.0001, TolAng=0.01, TolCurv=0.1, MaxDeg=8, MaxSegments=9):
+        """
+        Returns a plate surface that is 'thickness' thick, enclosed by 'surf_edge_pts' points,  and going through 'surf_pts' points.
+        
+        :param surf_edges
+        :type 1 surf_edges: list of [x,y,z] float ordered coordinates
+        :type 2 surf_edges: list of unordered CadQuery wires
+        :param surf_pts = [] (uses only edges if [])
+        :type surf_pts: list of [x,y,z] float coordinates
+        :param thickness = 0 (returns 2D surface if 0)
+        :type thickness: float >= 0
+        :param Degree = 3 (OCCT default)
+        :type Degree: Integer >= 2
+        :param NbPtsOnCur = 15 (OCCT default)
+        :type: NbPtsOnCur Integer >= 15
+        :param NbIter = 2 (OCCT default)
+        :type: NbIterInteger >= 2
+        :param Anisotropie = False (OCCT default)
+        :type Anisotropie: Boolean
+        :param: Tol2d = 0.00001 (OCCT default)
+        :type Tol2d: float > 0
+        :param Tol3d = 0.0001 (OCCT default)
+        :type Tol3dReal: float > 0
+        :param TolAng = 0.01 (OCCT default)
+        :type TolAngReal: float > 0
+        :param TolCurv = 0.1 (OCCT default)
+        :type TolCurvReal: float > 0
+        :param MaxDeg = 8 (OCCT default)
+        :type MaxDegInteger: Integer >= 2 (?)
+        :param MaxSegments = 9 (OCCT default)
+        :type MaxSegments: Integer >= 2 (?)
+        """
+        
+        # POINTS CONSTRAINTS: List of (x,y,z) points provided
+        pts_array = [gp_Pnt(*pt) for pt in surf_pts]
+        
+        # EDGE CONSTRAINTS: build closed polygon
+        if isinstance(surf_edges, list): # List of (x,y,z) points provided
+            e_array = [Vector(*e) for e in surf_edges]
+            wire_builder = BRepBuilderAPI_MakePolygon()
+            for e in e_array: # Create polygon from edges
+                wire_builder.Add(e.toPnt())
+            wire_builder.Close()
+            w = wire_builder.Wire()
+        else: # Closed wires provided
+            w = surf_edges
+        edges = [i for i in TopologyExplorer(w).edges()]
+        
+        # MAKE SURFACE
+        continuity = GeomAbs_C0 # Fixed, changing to anything else crashes.
+        face = Face.makeNSidedSurface(edges, pts_array, continuity, Degree, NbPtsOnCur, NbIter, Anisotropie, Tol2d, Tol3d, TolAng, TolCurv, MaxDeg, MaxSegments) 
+        
+        if abs(thickness)>0: # Thicken surface, abs() because negative values are allowed (sets direction of thickening)
```

Comment:
Is this really needed? Maybe an error (even from OCC) would be better?

Reply:
Here the idea is that ``BRepOffset_MakeOffset()`` accepts as offset negative and positive values, it just changes the direction of offsetting. The ``abs()`` is just here to make sure that the thickening of the surface is performed in both cases when thickness is different than 0. 
I modified the docstring ``:type thickness: float (may be negative or positive depending on thicknening direction)``
So in my opinion there would be no need to trigger an error. Is that right?